linked-lists/c-or-cpp/reverse-linkedlist.cpp

```diff
@@ -0,0 +1,70 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+struct Node
+{
+    int data;
+    Node *next;
+};
+
+Node *newNode(int data)
+{
+    Node *new_node = new Node;
+    new_node->data = data;
+    new_node->next = NULL;
+    return new_node;
+}
+
+void push(Node **head_ref, int new_data)
+{
+    Node *new_node = new Node();
+    new_node->data = new_data;
+    new_node->next = (*head_ref);
+    (*head_ref) = new_node;
+}
+
+Node *reverseList(Node *A)
+{
+    Node *current, *prev, *next;
+    current = A;
+    prev = NULL;
+    next = NULL;
+    while (current != NULL)
+    {
+
+        // storing address of node next to current
+        next = current->next;
+
+        // reversing the link of current node
+        current->next = prev;
+
+        // updating prev and current
+        prev = current;
+        current = next;
+    }
+
+    current = prev;
+
+    return current;
+}
+
+void solve()
+{
+    Node *head = NULL;
+    push(&head, 5);
+    push(&head, 7);
+    push(&head, 10);
+    reverseList(head);
+}
```

Comment:
No method present to print the result somewhere to show if the function actually did something or not. 

fix : - 
either create a different function to print the list before and after reversal to show the visual representation of working of the function or just add some likes of code doing the same right here

Reply:
I have added a function to print LinkedList before and after running the reversing function, please check :)